addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,408 +24,597 @@
+Green = "\\cs(0,255,0)"
+Red = "\\cs(255,0,0)"
+Yellow = "\\cs(255,255,0)"
+
+current_mob = nil
+current_proc = nil
+obj_time = 0
+end_time = 0
+
+window = texts.new(" ",settings.window,settings)
+w = T{}
+
+-------------------------------------------------------------------------------
+-- Initialize the Currency array. We need this to keep track of the drops -----
+-------------------------------------------------------------------------------
+function init_currency()
+    Currency = T{"Ordelle Bronzepiece","Montiont Silverpiece","One Byne Bill",
+                "One Hundred Byne Bill","Tukuku Whiteshell","Lungo-Nango Jadeshell",
+                "Forgotten Thought","Forgotten Hope","Forgotten Touch","Forgotten Journey","Forgotten Step"}
+    for i=1, #Currency do
+        w[Currency[i]] = 0
+    end
+end
+init_currency()
+
+-------------------------------------------------------------------------------
+-- Initialize the time Granules array. Keeps track of the time extensions -----
+-------------------------------------------------------------------------------
+function init_granules()
+    Granules = T{"Crimson granules of time","Azure granules of time","Amber granules of time",
+                "Alabaster granules of time","Obsidian granules of time"}
+    for i=1, #Granules do
+        w[Granules[i]] = 0
+    end
+end
+init_granules()
+
+-------------------------------------------------------------------------------
+-- The on screen window structure ---------------------------------------------
+-------------------------------------------------------------------------------
+function init_window()
+    local showCurrencyDivider = false
+    window.text(window,Yellow.."Time remaining: ${time|initializing...}")
+    window.appendline(window,"\\cr————————————————————")
+    window.appendline(window,"${current_mob|(unknown)}\n"..Green.."${current_proc|(none)}")
+    window.appendline(window,"\\cr————————————————————")
+    for i=1, #Currency do
```

Comment:
See above. As you're using the Currency table now it doesn't need to be a T-table at all.